Sage/carbon
# fix(toast): ensure closed toasts do not trigger axe warnings

fix #6923

### Proposed behaviour


Set "aria-hidden" on Toast region while its closed. 
Only define the "aria-labelledby" when the Toast is open. 

### Current behaviour


When a Toast's content is only defined on open, its accessible name cannot be computed and is "empty" while closed. This causes an axe issue when more than one Toast is rendered on a page with undefined content as their accessible name is not "unique".

![image](https://github.com/user-attachments/assets/b6e84bad-1b01-41f4-a4e7-f876ecd927e5)

### Checklist




#### QA


### Additional context



### Testing instructions



[Sandbox demonstrating issue](https://stackblitz.com/edit/parsium-carbon-starter-slnd56?file=src%2FApp.tsx)
- Run axe scan with both Toasts closed. 
- See issue shown in screenshot. 

Screen reader behaviour should not change with fix.
